pages/agent/v3/hooks.md.erb

```diff
@@ -129,12 +133,15 @@ they are run as part of each job:
 
 ### Creating job lifecycle hooks
 
-Job lifecycle hooks run for every job an agent accepts. They are shell
-scripts you can use to run commands and export environment variables. These
-hooks have access to all the standard [Buildkite environment variables](/docs/pipelines/environment-variables).
+Job lifecycle hooks are sourced for every job an agent accepts.
+Use job lifecycle hooks to prepare for jobs, override default behavior, or clean up after jobs that have finished.
+For example, use `environment` to set a job's environment variables or `pre-exit` to delete temporary files and remove containers.
+If your hook is related to the startup or shutdown of the agent, consider [agent lifecycle hooks](#job-lifecycle-hooks) for those tasks instead.
 
-Job lifecycle hooks are copied to `$TMPDIR` directory and *sourced* by the
-agent's default shell. This has a few implications:
+Job lifecycle hooks are shell scripts that you can use to run commands and export environment variables to jobs.
```

Comment:
Should this go before some of the other description?

Reply:
This whole thing felt a little half baked. Revised with 5be4f71153d0b07b7437f623e9454a73da507b83.